mlrun/frameworks/_common/artifacts_library.py

```diff
@@ -18,80 +18,165 @@ class 'init_artifact' class method:
     some_artifact = SomeArtifactPlan
     """
 
+    # A constant name for the context parameter to use for passing a plans configuration:
+    CONTEXT_PARAMETER = "_artifacts"
+
+    # TODO: Finish support for custom plans.
     @classmethod
-    def from_dict(cls, plans_dictionary: Dict[str, dict]) -> List[Plan]:
+    def get_plans(
+        cls,
+        artifacts: Union[List[Plan], Dict[str, dict], List[str]] = None,
+        context: mlrun.MLClientCtx = None,
+        include_default: bool = True,
+        # custom_plans: dict = None, :param custom_plans: Custom user plans objects to initialize from.
+        **default_kwargs,
+    ) -> List[Plan]:
+        """
+        Get plans for a run. The plans will be taken from the provided artifacts / configuration via code, from provided
+        configuration via MLRun context and if the 'include_default' is True, from the framework artifact library's
+        defaults.
```

Comment:
I did saw later down the code more Plan related stuff, but I am not quite sure why artifacts can also be a List[Plan]

Reply:
The idea for a plan is to produce an artifact before, during or post training / evaluation, hence thee name - artifact plan. But I decided to kill this feature later on. I will replace them with a class I shall name Evaluator or Analyzer. It will take a model and some parameters and will yield the artifacts and metrics it needs. It will be much easier to use as the more I see how users use MLRun, it will be more fitting.